tensorflow/compiler/xla/service/hlo_module_util.cc

```diff
@@ -129,8 +129,11 @@ StatusOr<std::unique_ptr<HloModuleConfig>> CreateModuleConfig(
         DeviceAssignment::Deserialize(execution_options->device_assignment()));
     config->set_static_device_assignment(*device_assignment);
   }
-  config->set_alias_passthrough_params(
```

Comment:
On line 97 it assumes that execution_options is non-null already, so I think we should remove the guard at line 125 instead (and probably your analyzer should highlight this guard as dead code instead, as it is guaranteed to be non-null as we reach it)

Reply:
Line 97 or 87? If you mean line 87, then it only ensures that execution_options is non-null up to the line 116. After that there is else-statement on lines 116-119 where control-flow is not interrupted. So execution_options still may be null on 125 and 132, unless I missed something

Reply:
Ah I see, thanks! Maybe then move both set_static_device_assignment and set_alias_passthrough_param under the guard starting at line 87 then?

Reply:
Yeah, it would be more neat solution